packages/dds/tree/src/test/simple-tree/schemaFactoryRecursive.spec.ts

```diff
@@ -166,6 +174,79 @@ describe("SchemaFactory Recursive methods", () => {
 			);
 		});
 
+		it("object with required recursive field", () => {
+			class ObjectRecursive extends sf.objectRecursive("Object", {
+				x: sf.requiredRecursive([() => ObjectRecursive, sf.number]),
+			}) {}
+			{
+				type _check = ValidateRecursiveSchema<typeof ObjectRecursive>;
+			}
+
+			type XSchema = typeof ObjectRecursive.info.x;
+			type Field2 = XSchema extends FieldSchema<infer Kind, infer Types>
+				? ApplyKind<TreeNodeFromImplicitAllowedTypes<Types>, Kind, false>
+				: "zzz";
+			type XTypes = XSchema extends FieldSchemaUnsafe<infer Kind, infer Types> ? Types : "Q";
+			type Field3 = TreeNodeFromImplicitAllowedTypes<XTypes>;
+			type Field4 = FlexListToUnion<XTypes>;
+			type _check1 = requireTrue<areSafelyAssignable<Field3, ObjectRecursive | number>>;
+			type _check2 = requireTrue<
+				areSafelyAssignable<
+					Field4,
+					| TreeNodeSchemaClass<
+							"com.fluidframework.leaf.number",
+							NodeKind.Leaf,
+							number,
+							number
+					  >
+					| TreeNodeSchemaNonClass<
+							"com.fluidframework.leaf.number",
+							NodeKind.Leaf,
+							number,
+							number
+					  >
+					| typeof ObjectRecursive
```

Comment:
We should avoid hard coding qualified node names. numberSchema.name or schema.number.name.

Here though you can do even further:

```suggestion
					typeof schema.number | typeof ObjectRecursive
```

Reply:
Also its really strange to manually expand TreeNodeSchema into TreeNodeSchemaClass or TreeNodeSchemaNonClass. Perhaps an extra inlining we don't need lead to that? Did something actually display that being split/inlined somewhere?

Reply:
Yeah, intellisense showed that when I hovered over `Field4`